cfme/tests/infrastructure/test_host.py

```diff
@@ -529,6 +536,45 @@ def test_add_ipmi_refresh(appliance, setup_provider):
     assert view.ipmi_address.read() == ipmi_address
 
 
+@pytest.fixture
+def host(appliance, provider):
+    host_collection = provider.hosts
+    return random.choice(host_collection.all())
+
+
+HOST_DETAIL_ROWS = [
+    ("Properties", "VMM Information", HostVmmInfoView),
+    ("Properties", "Operating System", HostOsView),
+    ("Properties", "Devices", HostDevicesView),
+    ("Properties", "Networks", HostNetworkDetailsView),
+    ("Properties", "Storage Adapters", HostStorageAdaptersView),
+    ("Configuration", "Services", HostServicesView),
+    ("Toolbar", "Print or export", HostPrintView),
+]
+
+
+@test_requirements.infra_hosts
+@pytest.mark.parametrize("table,row,view", HOST_DETAIL_ROWS,
+    ids=[rel[1] for rel in HOST_DETAIL_ROWS])
+def test_infrastructure_hosts_viewing(appliance, setup_provider, host, table, row, view):
+    """
+    Polarion:
+        assignee: prichard
+        casecomponent: Infra
+        caseimportance: high
+        initialEstimate: 1/6h
+    """
+    # Except for OS, Network, and print we need to check if we have any of the row items.
+    if row not in ["Operating System", "Networks", "Print or export"]:
+        view = navigate_to(host, "Details")
+        if view.entities.summary(table).get_text_of(row) == '0':
+            return
```

Comment:
This will show up as a `pass` afaik, we should skip the test in this case I think. I.e.
```suggestion
            pytest.skip(f"No row item present for {row}")
```